UnicornRender/video/include/unicorn/video/Mesh.hpp

```diff
@@ -9,76 +9,94 @@
 
 #include <unicorn/utility/SharedMacros.hpp>
 #include <unicorn/video/Material.hpp>
+#include <unicorn/video/Transform.hpp>
 
 #include <wink/signal.hpp>
 #include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <glm/gtc/matrix_transform.hpp>
 
 #include <list>
 
 namespace unicorn
 {
 namespace video
 {
-/**
- * @brief Vertex information
- */
+
+/** @brief Vertex information */
 struct Vertex
 {
-    /**
-     * @brief Position of vertex
-     */
+    /** @brief Position of vertex */
     glm::vec3 pos;
 
-    /**
-    * @brief Texture coordinates of vertex
-    */
+    /** @brief Texture coordinates of vertex */
     glm::vec2 tc;
 };
 
-/**
-* @brief Mesh data
-*/
-class Mesh
+/** @brief Mesh data */
+class Mesh : public Transform
 {
 public:
+    /**
+     * @brief Constructs mesh
+     *
+     * @param[in] material visual describing material
+     */
     UNICORN_EXPORT Mesh(Material const& material);
+
+    /** @brief Default destructor */
     UNICORN_EXPORT ~Mesh() = default;
 
     /**
     * @brief Updates vertices and indices geometry
-    * @param [in] vertices vertexes data
-    * @param [in] indices indices data
+     *
+    * @param[in] vertices vertices data
+    * @param[in] indices indices data
     */
     UNICORN_EXPORT void SetMeshData(std::vector<Vertex> const& vertices, std::vector<uint32_t> const& indices);
 
     /**
     * @brief Sets new material
-    * @param [in] material updated material for current mesh
+    *
+    * @param[in] material updated material for current mesh
     */
     UNICORN_EXPORT void SetMaterial(Material const& material);
 
     /**
      * @brief Returns mesh vertices
+     *
      * @return Mesh vertices
      */
     UNICORN_EXPORT const std::vector<Vertex>& GetVertices() const;
 
     /**
     * @brief Returns mesh indices
+    *
     * @return Mesh indices
     */
     UNICORN_EXPORT std::vector<uint32_t> const& GetIndices() const;
 
     /**
     * @brief Returns mesh material
+    *
     * @return mesh material data
     */
     UNICORN_EXPORT Material const& GetMaterial() const;
 
     /**
-    * @brief Matrix for model transformations
-    */
-    glm::mat4 modelMatrix;
+     * @brief Scales mesh
+     *
+     * @param[in] scale Each axis scale factor
+     *
+     * {1,1,1} is origin scale, {0.5, 0.5, 2} is x and y diminished twice
```

Comment:
`Lineary scales each basis component of the mesh vectors by the given scale factors.
Example:`